include/recognition/CographRecognition.hpp

```diff
@@ -37,4 +44,65 @@ class CographRecognition : public NetworKit::Algorithm {
     NetworKit::Graph graph;
 };
 
+/**
+ * @ingroup recognition
+ * The class for recognition of cographs procedure from
+ * Bretscher, Corneil, Habib, Paul, "A Simple Linear Time LexBFS Cograph Recognition Algorithm".
+ *
+ */
+class BretscherCorneilHabibPaulCographRecognition : public CographRecognition {
+ public:
+    using CographRecognition::CographRecognition;
+    std::pair<std::vector<std::vector<std::pair<int, int>>>, std::vector<NetworKit::node>>
+    LexBfsMinus(bool is_complement, std::vector<NetworKit::node> &a);
+    bool NeighbourhoodSubsetProperty(bool is_complement, std::vector<NetworKit::node> a,
+    std::vector<std::vector<std::pair<int, int>>> borders);
+    void run() override;
+    bool isCograph() const override;
+ private:
+    bool is_cograph = false;
+};
+
+/**
+ * @ingroup recognition
+ * The class for recognition of cographs procedure from
+ * Corneil, Stewart, Perl, "A linear recognition algorithm for cographs".
+ *
+ */
+class CorneilStewartPerlCographRecognition : public CographRecognition {
+ public:
+    using CographRecognition::CographRecognition;
+    enum class State {
+        UNKNOWN,
+        COGRAPH,
+        CONTAINS_0_NODE,
+        EXISTS_1_NODE_NOT_PROPERLY_MARKED,
+        GRANDPARENT_IS_NOT_IN_SET,
+        NO_ONE_PATH,
+        WRONG_PARENT,
+        WRONG_GRANDPARENT
+    };
+    /**
+     * Return the graph type found by the algorithm.
+     *
+     * @return State of the graph.
+     */
+    State getState() const;
+    CorneilStewartPerlCographRecognition::State Recognition();
+    void run() override;
+    bool isCograph() const override;
+    void Unmark();
+    void Mark(CoNode *x);
+    std::pair<CoNode*, CorneilStewartPerlCographRecognition::State>FindLowest() const;
+    void InsertXToCoTree(CoNode *u, CoNode *x);
```

Comment:
Why can't they be private? If CoTree needs it, consider friending this class there (or other way round).

Reply:
fixed